chingu-coders/Voyage2-Turtles-02
# stickynotes bug fixes and refactor

## Purpose
* Fix several bugs in stickynotes
* refactor for cleaner, more readable, more performant code

## Approach/Fixes
* The main issue was that changes to stickynotes were not persisting across reloads, and there was strange behavior like random changes in state and deleted notes coming back (ZOMBIE NOTES). I identified that the problem was that a single reference to local storage was being stored in the closure of all event holders, meaning that when state would change, it would not update across all event listeners attached to the stickynote DOM element.

## Testing
* Check out this PR
* create a few sticky notes
* change all possible parameters. reload and confirm that changes persist
* delete a few notes, reload and confirm that they stay deleted
